rollpe/heart/tests.py

```diff
@@ -11,53 +11,19 @@
 from heart.models import Heart
 
 
-"""
-    마음 조회 시나리오
-    
-    - 공통
-    1. 로그인 x => 마음목록 조회 불가능
-    
-    - 공개 롤링 페이퍼 => 
-        2. 로그인 o => 조회 가능
-        
-    - 비공개 롤링 페이퍼 =>
-        3. 로그인 o, 초대된 유저 x => 조회 불가능
-        4. 로그인 o, 초대된 유저 o => 조회 가능
-        
-        
-    - 인덱스 , 블러처리 필요
-
-"""
-class HeartReadAPITest(APITestCase):
+class BaseTestSetup(APITestCase):
     @classmethod
     def create_test_users(cls):
         """테스트 유저 생성"""
-        cls.receiver = User.objects.create(
-            name='receiver',
-            email='receiver@gmail.com',
-        )
-        cls.host = User.objects.create(
-            name='host',
-            email='host@gmail.com',
-        )
-        cls.user1 = User.objects.create(
-            name='test_user1',
-            email='testuser1@gmail.com',
-        )
-        cls.user2 = User.objects.create(
-            name='test_user2',
-            email='testuser2@gmail.com',
-        )
-        cls.user3 = User.objects.create(
-            name='test_user3',
-            email='testuser3@gmail.com',
-        )
-        cls.user1.set_password('1234')
-        cls.user1.save()
-        cls.user2.set_password('1234')
-        cls.user2.save()
-        cls.user3.set_password('1234')
-        cls.user3.save()
+        cls.receiver = User.objects.create(name='receiver', email='receiver@gmail.com')
+        cls.host = User.objects.create(name='host', email='host@gmail.com')
+        cls.user1 = User.objects.create(name='test_user1', email='testuser1@gmail.com')
+        cls.user2 = User.objects.create(name='test_user2', email='testuser2@gmail.com')
+        cls.user3 = User.objects.create(name='test_user3', email='testuser3@gmail.com')
+
+        for user in [cls.user1, cls.user2, cls.user3]:
+            user.set_password('1234')
+            user.save()
```

Comment:
Django 에서 지원하는 bulk_create()를 사용할 순 없을까요 ?
```suggestion
            User.objects.bulk_create(user_list)
```

Reply:
좋은 생각!

Reply:
search 해보니 set_password()는 비밀번호를 해싱해서 저장하는 과정이라
일반적인 ORM보다 상위 레벨이라 bulk_create()를 적용할 수 없다고 해요.

Reply:
`
users = [
            User(name='receiver', email='receiver@gmail.com'),
            User(name='host', email='host@gmail.com'),
            User(name='test_user1', email='testuser1@gmail.com'),
            User(name='test_user2', email='testuser2@gmail.com'),
            User(name='test_user3', email='testuser3@gmail.com'),
        ]
        User.objects.bulk_create(users)

        # 생성된 객체 다시 가져오기
        cls.receiver, cls.host, cls.user1, cls.user2, cls.user3 = User.objects.all()[:5]

        # 비밀번호 암호화 후 업데이트
        User.objects.filter(id__in=[cls.user1.id, cls.user2.id, cls.user3.id]).update(
            password=make_password('1234')
        )
`

따라서 기존보다 더 나은 로직을 구현 하기 위해 해당 방법으로 리팩토링 했습니다.